symphonia-core/src/units.rs

```diff
@@ -118,6 +122,13 @@ impl From<f64> for Time {
     }
 }
 
+impl From<StdDuration> for Time {
+    fn from(duration: StdDuration) -> Self {
+        let seconds = duration.as_secs_f64();
+        Time::new(seconds.trunc() as u64, seconds.fract())
```

Comment:
Same idea about improving precision by removing the round-trip through a `f64`.

```suggestion
        Time::new(duration.as_secs(), f64::from(duration.subsec_nanos()) / 1_000_000_000.0)
```

Reply:
Done.